sbin/prepareWorkspace.sh

```diff
@@ -577,6 +564,67 @@ prepareMozillaCacerts() {
     fi
 }
 
+# Create and setup GNUPGHOME
+setupGpg() {
+    ## This affects riscv64 & Alpine docker images and also evaluation pipelines
+    if ( [ -r /etc/alpine-release ] && [ "$(pwd | wc -c)" -gt 83 ] ) || \
+       ( [ "${BUILD_CONFIG[OS_KERNEL_NAME]}" == "linux" ] && [ "${BUILD_CONFIG[OS_ARCHITECTURE]}" == "riscv64" ] && [ "$(pwd | wc -c)" -gt 83 ] ); then
+        # Use /tmp in preference to $HOME as fails gpg operation if PWD > 83 characters
+        # Also cannot create ~/.gpg-temp within a docker context
+        GNUPGHOME="$(mktemp -d /tmp/.gpg-temp.XXXXXX)"
+    else
+        GNUPGHOME="${BUILD_CONFIG[WORKSPACE_DIR]:-$PWD}/.gpg-temp"
+    fi
+    if [ ! -d "$GNUPGHOME" ]; then
+        mkdir -m 700 "$GNUPGHOME"
+    fi
+    export GNUPGHOME
+
+    echo "GNUPGHOME=$GNUPGHOME"
+}
+
+# Download the required DevKit if necessary
+downloadDevkit() {
+  if [[ -n "${BUILD_CONFIG[USE_ADOPTIUM_DEVKIT]}" ]]; then
+    rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/devkit"
+    mkdir -p "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/devkit"
+
+    devkit_tar="${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/devkit/devkit.tar.xz"
+
+    setupGpg
+
+    # Determine DevKit tarball to download for this arch and release
+    devkitUrl="https://github.com/adoptium/devkit-binaries/releases/download/${BUILD_CONFIG[USE_ADOPTIUM_DEVKIT]}"
+    devkit_target="${BUILD_CONFIG[OS_ARCHITECTURE]}-linux-gnu"
+    devkit="devkit-$(echo "${BUILD_CONFIG[USE_ADOPTIUM_DEVKIT]}" | cut -d'-' -f1-3)-${devkit_target}-$(echo "${BUILD_CONFIG[USE_ADOPTIUM_DEVKIT]}" | cut -d'-' -f4)"
```

Comment:
Is there a reason why we haven't made the filename a real substring of the release name? This manuipulation seems to add additional unnecessary complexity for retrieving the appropriate one with a parameter.

Reply:
so I assume you mean the "reverse" ? as the filename contains the "arch", and the release contains DevKit's for all the arch's ?
Maybe easier to view an example : https://github.com/adoptium/devkit-binaries/releases/tag/gcc-11.3.0-Centos7.9.2009-b01
The filename is a "superset" of the release tag

Reply:
so the above code needs to get the correct "architecture" tarball for the given release

Reply:
> so I assume you mean the "reverse"

Correct 😳 

>  The filename is a "superset" of the release tag

🤔 
**Release tag:** gcc-11.3.0-Centos7.9.2009-b01
**Filename:** devkit-gcc-11.3.0-Centos7.9.2009-aarch64-linux-gnu-b01.tar.xz

The `b01` is not in the same place, so you have to insert another bit in the middle of the release tag to generate the filename instead of it just being `devkit-${tag}-${architecture}-linux-gnu.tar.xz

Reply:
yeah, i'm just used to putting "build numbers" at the end of things :-)
I can switch it around as you suggest, sounds sensible, cheers

Reply:
done